client/my-sites/email/titan-mail-quantity-selection/index.jsx

```diff
@@ -203,14 +212,126 @@ class TitanMailQuantitySelection extends React.Component {
 						}
 					) }
 				</span>
+			</CompactCard>
+		);
+	}
+
+	doesAdditionalPriceMatchStandardPrice() {
+		const { selectedDomain, titanMonthlyProduct } = this.props;
+		if ( ! selectedDomain || ! hasTitanMailWithUs( selectedDomain ) ) {
+			return true;
+		}
+		const costPerAdditionalMailbox = getTitanMailboxPurchaseCost( selectedDomain );
+		if ( ! costPerAdditionalMailbox ) {
+			return true;
+		}
+		return (
+			costPerAdditionalMailbox.amount === titanMonthlyProduct.cost &&
+			costPerAdditionalMailbox.currency === titanMonthlyProduct.currency_code
+		);
+	}
+
+	renderPricingDetails() {
+		const { moment, selectedDomain, titanMonthlyProduct, translate } = this.props;
+
+		const pricingTitle = <h3>{ translate( 'Pricing' ) }</h3>;
+		// Handle cases where we're only dealing with the standard price
+		if ( this.doesAdditionalPriceMatchStandardPrice() ) {
+			return (
+				<React.Fragment>
+					{ pricingTitle }
+					<span>
+						{ translate( 'Each mailbox costs {{strong}}%(price)s/month{{/strong}}', {
+							args: {
+								price: titanMonthlyProduct.cost_display,
+							},
+							components: {
+								strong: <strong />,
+							},
+							comment:
+								'%(price)s is a formatted price for each mailbox, e.g. $3.50, €3.75, or PLN 3.75',
+						} ) }
+					</span>
+				</React.Fragment>
+			);
+		}
+
+		const purchaseCost = getTitanMailboxPurchaseCost( selectedDomain );
+		const renewalCost = getTitanMailboxRenewalCost( selectedDomain );
+		const expiryDate = getTitanExpiryDate( selectedDomain );
+
+		const prorationMessage =
+			purchaseCost.amount < renewalCost.amount
+				? translate(
+						'Your initial purchase price is less than your standard monthly price of %(price)s per mailbox because ' +
+							"we're only charging you for the remainder of the current month.",
+						{
+							args: {
+								price: renewalCost.text,
+							},
+							comment:
+								'%(price)s is a formatted price for a monthly email mailbox subscription, e.g. $3.50, €3.75, or PLN 4.50',
+						}
+				  )
+				: translate(
+						'Your initial purchase price is more than your standard monthly price of %(price)s per mailbox because ' +
+							"we're charging you for the remainder of the current month plus one or more months of service.",
```

Comment:
Maybe we should use "as well as" instead of "plus"? The idea is that we're charging the user for the rest of the current month as well as "one or more months of service" until their expiry date. We can definitely iterate on this language!